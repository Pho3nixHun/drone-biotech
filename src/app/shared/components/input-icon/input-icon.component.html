<span class="label-text my-auto flex items-center">
    <mat-icon>{{ icon() }}</mat-icon>
</span>
