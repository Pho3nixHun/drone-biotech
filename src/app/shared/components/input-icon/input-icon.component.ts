import { Component, input } from '@angular/core';
import { MatIconModule } from '@angular/material/icon';
import { MatIcon } from '@interfaces/mat-icon.enum';

@Component({
    selector: 'app-input-icon',
    imports: [MatIconModule],
    templateUrl: './input-icon.component.html',
})
export class InputIconComponent {
    public icon = input.required<MatIcon>();
}
