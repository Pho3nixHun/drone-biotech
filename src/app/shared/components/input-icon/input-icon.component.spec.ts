import { ComponentFixture, TestBed } from '@angular/core/testing';

import { InputIconComponent } from './input-icon.component';
import { Component, signal } from '@angular/core';
import { MatIcon } from '@interfaces/mat-icon.enum';

@Component({
    imports: [InputIconComponent],
    template: `<app-input-icon [icon]="leadingIcon()" />`,
})
class TestHostComponent {
    public leadingIcon = signal<MatIcon>(MatIcon.ADD);
}
describe('InputIconComponent', () => {
    let fixture: ComponentFixture<TestHostComponent>;
    let compiled: HTMLElement;

    beforeEach(async () => {
        await TestBed.configureTestingModule({
            imports: [TestHostComponent],
        }).compileComponents();

        fixture = TestBed.createComponent(TestHostComponent);
        compiled = fixture.debugElement.nativeElement;
    });

    // Snapshot testing
    it('should render the template correctly', () => {
        // There is no need to arrange

        // Act
        fixture.detectChanges();

        // Assert
        expect(compiled).toMatchSnapshot();
    });
});
