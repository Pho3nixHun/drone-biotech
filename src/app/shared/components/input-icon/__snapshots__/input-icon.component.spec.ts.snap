// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`InputIconComponent should render the template correctly 1`] = `
<div
  id="root0"
  ng-version="20.1.7"
>
  <app-input-icon>
    <span
      class="label-text my-auto flex items-center"
    >
      <mat-icon
        aria-hidden="true"
        class="mat-icon notranslate material-icons mat-ligature-font mat-icon-no-color"
        data-mat-icon-type="font"
        role="img"
      >
        add
      </mat-icon>
    </span>
  </app-input-icon>
</div>
`;
